rust/lance/src/dataset/fragment.rs

```diff
@@ -504,10 +504,13 @@ impl FileFragment {
 
         // scan with predicate and row ids
         let mut scanner = self.scan();
-        scanner
-            .with_row_id()
-            .filter(predicate)?
-            .project::<&str>(&[])?;
+        scanner.with_row_id();
+
+        // if predicate is true, delete the whole fragment
+        // else, filter the predicate
+        if predicate != "true" {
```

Comment:
This handles:

```
dataset.delete("true")
```

But not:

```python
dataset.delete("TRUE")
dataset.delete(" true")
```

both of which are technically the same thing.

Could you trim and lower for this comparison? (Also switching to equals, see next comment)

```suggestion
        if &predicate.trim().to_lowercase() == "true" {
```